decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```

Comment:
- [ ] Please make sure the `nameable` variable is readable and writable for the associations to work as expected.
  - You should define the `nameable` variable in attr_accessor. `attr_accessor: nameable`.